src/annotator/notebook.js

```diff
@@ -57,23 +72,39 @@ export default class Notebook extends Delegator {
     if (needIframe) {
       this.frame?.remove();
       this.frame = createNotebookFrame(this.options, this._groupId);
-      this.container.appendChild(this.frame);
-      this.element.appendChild(this.container);
+      container.appendChild(this.frame);
     }
   }
 
   show() {
+    const container = this._initContainer();
     this._update();
-    this.container.classList.add('is-open');
-    this.container.style.display = '';
+    container.classList.add('is-open');
+    container.style.display = '';
   }
 
   hide() {
-    this.container.classList.remove('is-open');
-    this.container.style.display = 'none';
+    if (this.container) {
+      this.container.classList.remove('is-open');
+      this.container.style.display = 'none';
+    }
   }
 
   destroy() {
-    this.frame?.remove();
+    this._outerContainer.remove();
+  }
+
+  _initContainer() {
```

Comment:
To be truthful I'm not quite sure how much resource usage is saved by lazily creating this container. The main possible benefit is not loading an extra copy of the annotator CSS inside the shadow root. I believe that some browsers are able to share parsed stylesheets between different shadow DOM elements. If we later learn that the stylesheet is effectively shared in most browsers, we might want to simplify this code a little by removing the lazy initialization.

Reply:
I agree.